v2/examples/kilroy/main.go

```diff
@@ -0,0 +1,209 @@
+// processed by the generators in this package.
+func getDefaultNameSystem() string {
+	return "raw"
+}
+
+// getTargets is called after the inputs have been loaded.  It is expected to
+// examine the provided context and return a list of Packages which will be
+// executed further.
+func getTargets(c *generator.Context, args *Args) []generator.Target {
+	boilerplate, err := gengo.GoBoilerplate(args.goHeaderFile, gengo.StdBuildTag, gengo.StdGeneratedBy)
+	if err != nil {
+		klog.Fatalf("failed loading boilerplate: %v", err)
+	}
+
+	targets := []generator.Target{}
+	for _, input := range c.Inputs {
+		klog.V(2).InfoS("processing", "pkg", input)
+
+		pkg := c.Universe[input]
+
+		targets = append(targets, &generator.SimpleTarget{
+			PkgName:       pkg.Name,
+			PkgPath:       pkg.Path, // output pkg is the same as the input
+			PkgDir:        pkg.Dir,  // output pkg is the same as the input
+			HeaderComment: boilerplate,
+
+			// FilterFunc returns true if this Package cares about this type.
+			// Each Generator has its own Filter method which will be checked
+			// subsequently.  This will be called for every type in every
+			// loaded package, not just things in our inputs.
+			FilterFunc: func(c *generator.Context, t *types.Type) bool {
+				return t.Name.Package == pkg.Path
+			},
+
+			// GeneratorsFunc returns a list of Generators, each of which is
+			// responsible for a single output file (though multiple generators
+			// may write to the same one).
+			GeneratorsFunc: func(c *generator.Context) (generators []generator.Generator) {
+				return []generator.Generator{
+					newKilroyGenerator(args.outputFile, pkg, args.methodName),
+				}
+			},
+		})
+	}
+
+	return targets
+}
+
+// kilroyGenerator produces a file with autogenerated functions.
+type kilroyGenerator struct {
+	generator.GolangGenerator
+	myPackage  *types.Package
+	methodName string
+}
+
+func newKilroyGenerator(outputFilename string, pkg *types.Package, methodName string) generator.Generator {
+	return &kilroyGenerator{
+		GolangGenerator: generator.GolangGenerator{
+			OutputFilename: outputFilename,
+		},
+		myPackage:  pkg,
+		methodName: methodName,
+	}
+}
+
+// Filter returns true if this Generator cares about this type.
+// This will be called for every type which made it through this Package's
+// Filter method.
+func (g *kilroyGenerator) Filter(c *generator.Context, t *types.Type) bool {
+	// We only handle exported structs.
+	return t.Kind == types.Struct && !namer.IsPrivateGoName(t.Name.Name)
+}
+
+// Namers returns a set of NameSystems which will be merged with the namers
+// provided when executing this package. In case of a name collision, the
+// values produced here will win.
+func (g *kilroyGenerator) Namers(*generator.Context) namer.NameSystems {
+	return namer.NameSystems{
+		// This elides package names when the name is in "this" package.
+		"raw": namer.NewRawNamer(g.myPackage.Path, nil),
```

Comment:
Maybe this works the same as previous gengo but I'm noticing this shadows the `raw` name passed via `getNameSystems` to `gengo.Execute`. How do the two namesystems interact and handle conflicts?

Reply:
I took a poke at this and realized that some of the tools use the Target-scope namers to avoid allocating the exact same namers for each Generator.  We could pass a `NameSystems` and a `Namer` into `Execute` (instead of a `NameSystems` and a `string`) but it doesn't save anything.